cocos/core/geometry/geometry-native-ext.ts

```diff
@@ -87,7 +87,10 @@ const defineAttrFloat = (kls: Constructor, attr: string) => {
     // __nativeFields__ is defined in jsb_geometry_manual.cpp
     const desc: FieldDesc = (kls as any).__nativeFields__[attr];
     const cacheKey = `_$_${attr}`;
-    console.assert(desc.fieldSize === 4, `field ${attr} size ${desc.fieldSize}`);
+    if (!window.oh) {
+        // openharmony does not support the console.assert interface at this time.
+        console.assert(desc.fieldSize === 4, `field ${attr} size ${desc.fieldSize}`);
```

Comment:
How about implementing the `console.assert` in ScriptEngine.cpp and export it to JS?
Then, this `if (!window.oh)` check could be removed.